slack-lint-checks/src/main/java/slack/lint/retrofit/RetrofitUsageDetector.kt

```diff
@@ -39,11 +39,15 @@ class RetrofitUsageDetector : Detector(), SourceCodeScanner {
           HTTP_ANNOTATIONS.firstNotNullOfOrNull { node.findAnnotation(it) } ?: return
 
         val returnType = node.safeReturnType(context)
-        if (
-          returnType == null ||
-            returnType == PsiTypes.voidType() ||
-            returnType.canonicalText == "kotlin.Unit"
-        ) {
+        val isSuspend = context.evaluator.isSuspend(node)
+        val hasAllowUnitResponseAnnotation = node.findAnnotation("slack.lint.annotations.AllowUnitResponse") != null
```

Comment:
If this is false we should be able to skip needing the isVoidOrUnitReturnType variable right? Let's make that logic lazier